cmd/cloudtrail/write-events.go

```diff
@@ -0,0 +1,325 @@
+		pages += pageIter
+	}
+
+	maxPages := pages
+	for counter := 0; counter <= maxPages; counter++ {
+
+		lookupOutput, err := cloudtailClient.LookupEvents(ctx, &input)
+		if err != nil {
+			return nil, err
+		}
+
+		allookupOutputs = append(allookupOutputs, lookupOutput)
+
+		input.NextToken = lookupOutput.NextToken
+		if lookupOutput.NextToken == nil {
+			break
+		}
+
+	}
+
+	return allookupOutputs, nil
+}
+
+// Extracts Raw cloudtrailEvent Details
+func extractUserDetails(cloudTrailEvent *string) (*RawEventDetails, error) {
+	if cloudTrailEvent == nil || *cloudTrailEvent == "" {
+		return &RawEventDetails{}, fmt.Errorf("cannot parse a nil input")
+	}
+	var res RawEventDetails
+	err := json.Unmarshal([]byte(*cloudTrailEvent), &res)
+	if err != nil {
+		return &RawEventDetails{}, fmt.Errorf("could not marshal event.CloudTrailEvent: %w", err)
+	}
+
+	const supportedEventVersionMajor = 1
+	const minSupportedEventVersionMinor = 8
+
+	var responseMajor, responseMinor int
+	if _, err := fmt.Sscanf(res.EventVersion, "%d.%d", &responseMajor, &responseMinor); err != nil {
+		return &RawEventDetails{}, fmt.Errorf("failed to parse CloudTrail event version: %w", err)
+	}
+	if responseMajor != supportedEventVersionMajor || responseMinor < minSupportedEventVersionMinor {
+		return &RawEventDetails{}, fmt.Errorf("unexpected event version (got %s, expected compatibility with %d.%d)", res.EventVersion, supportedEventVersionMajor, minSupportedEventVersionMinor)
+	}
+	return &res, nil
+}
+
+// GenerateLink generates a hyperlink for the given URL and display text based of value pairs in cloudTrail Event.
+func generateLink(raw RawEventDetails) (url_link string) {
+	str1 := "https://"
+	str2 := ".console.aws.amazon.com/cloudtrailv2/home?region="
+	str3 := "#/events/"
+
+	eventRegion := raw.EventRegion
+	eventId := raw.EventId
+
+	var url = str1 + eventRegion + str2 + eventRegion + str3 + eventId
+	url_link = url
+
+	return url_link
+}
+
+// Join all individual patterns into a single string separated by the "|" operator
+func mergeRegex(regexlist []string) string {
+	return strings.Join(regexlist, "|")
+}
+
+// FilterUsers filters out events based on the specified Ignore list, which contains
+// regular expression patterns. It takes a slice of cloudtrail.LookupEventsOutput,
+// which represents the output of AWS CloudTrail lookup events operation, and a list
+// of regular expression patterns to ignore.
+func filterUsers(lookupOutputs []*cloudtrail.LookupEventsOutput, Ignore []string, shouldFilter bool) (*[]types.Event, error) {
+	filteredEvents := []types.Event{}
+	mergedRegex := mergeRegex(Ignore)
+
+	for _, lookupOutput := range lookupOutputs {
+		for _, event := range lookupOutput.Events {
+			raw, err := extractUserDetails(event.CloudTrailEvent)
+			if err != nil {
+				return nil, err
+			}
+			userArn := raw.UserIdentity.SessionContext.SessionIssuer.Arn
+			regexOdj := regexp.MustCompile(mergedRegex)
+			matchesUsrname := false
```

Comment:
```suggestion
			matchesUsername := false
```

Might as well type it out if it's only 1 character missing.